src/routes/view-invoice.tsx

```diff
@@ -0,0 +1,143 @@
+import { Alert, AlertTitle, Box } from '@mui/material';
+import { visuallyHidden } from '@mui/utils';
+import {
+  useParams,
+  useSearchParams,
+  useNavigate,
+  Link,
+} from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { useGetInvoice, useUpdateInvoice } from '../hooks/invoices.hooks.ts';
+import { z } from 'zod';
+import { InvoiceSchema } from '../types/Invoice.ts';
+
+import { InvoiceForm, InvoiceFormProps } from '../components/InvoiceForm.tsx';
+import Button from '@mui/material/Button';
+import Icon from '@mui/material/Icon';
+
+export const ViewInvoicePage = () => {
+  const { t } = useTranslation();
+  const { id } = z.object({ id: z.string() }).parse(useParams());
+  const { data: invoice } = useGetInvoice(id);
+  const navigate = useNavigate();
+
+  const [searchParams] = useSearchParams();
+
+  const isEditMode = searchParams.get('mode') === 'edit';
+
+  const updateInvoiceMutation = useUpdateInvoice();
+
+  const onSubmit = async (data: any) => {
+    if (!isEditMode || !id) return;
+
+    await updateInvoiceMutation.mutateAsync(
+      { id, data },
+      {
+        onSuccess: () => {
+          navigate(`/invoice/${id}`);
+        },
+        onError: () => {},
+      },
+    );
+  };
+
+  const formProps: InvoiceFormProps = {
+    defaultValues: invoice,
+    mode: isEditMode ? 'edit' : 'view',
+    resolver: InvoiceSchema,
+    onSubmit,
+    children: () => (
+      <ViewInvoiceActions
+        isEditMode={isEditMode}
+        id={id}
+        useMutation={updateInvoiceMutation}
+      />
+    ),
+  };
+
+  return (
+    <div>
+      <header>
+        <Box sx={{ m: 4 }}>
+          <h1 style={visuallyHidden}>{t('ADD_INVOICE')}</h1>
+        </Box>
+      </header>
+
+      <ViewInvoiceAlerts useMutation={updateInvoiceMutation} />
+
+      <InvoiceForm {...formProps} />
```

Comment:
You need to await and only render the component when data is fetched.

```ts
   {invoice ? <InvoiceForm {...formProps} /> : <Loading />}
```

Also I'd rather pass props explicitly in JSX, instead of creating intermediate variable like `formProps`